### PR TITLE
Add support for trimming collator data

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
Tested locally using build from @lewing's branch.

**Review by lewing — COMMENTED** [review]
Looks good

**Review by captainsafia — APPROVED** [review]

File: src/Components/WebAssembly/Build/src/targets/CollationLinkerDescriptor.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<linker>
+  <!-- This file disables exclusions that remove collation data -->
```
Comment (SteveSandersonMS) [inline src/Components/WebAssembly/Build/src/targets/CollationLinkerDescriptor.xml:2]: It's the opposite now isn't it? This file now *does* remove the collation data.

BTW please don't feel the need to rebuild this PR just for minor tweaks :) We can correct comments later.

File: src/Components/test/E2ETest/Tests/WebAssemblyStringComparisonTest.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        [Fact]
+        public void InvariantCultureWorksAsExpected()
+        {
+            Navigate(ServerPathBase, noReload: false);
+            Browser.MountTestComponent<StringComparisonComponent>();
+
+            var result = Browser.Exists(By.Id("results"));
+
+            Assert.Equal("Ordinal: False Invariant: True", result.Text);
```
Comment (SteveSandersonMS) [inline src/Components/test/E2ETest/Tests/WebAssemblyStringComparisonTest.cs:32]: I don't fully get how this test is passing *before* we take the Mono update that includes collation data by default. If this somehow works even without the Mono update, does that mean we should be using some other test case that requires the update?

Comment (SteveSandersonMS) [reply src/Components/test/E2ETest/Tests/WebAssemblyStringComparisonTest.cs:32]: My guess is that the collation data was included in the previous Mono update, and the one we were waiting for just changes details about how it works as per all the discussion on the Mono repo. If you have the full story I'd like to know :)

**Review by SteveSandersonMS — APPROVED** [review]
Nice!

I did add a couple of minor questions but have verified locally that this change does behave as stated (collation data included by default, but is excluded if you set `<BlazorWebAssemblyPreserveCollationData>false</BlazorWebAssemblyPreserveCollationData>`).

Based on that, I'm going to merge this right away. Hopefully that is a helpful thing to do but please let me know if this was an unwanted action :)

File: src/Components/test/E2ETest/Tests/WebAssemblyStringComparisonTest.cs (hunk 2 above)
Comment (pranavkm) [reply src/Components/test/E2ETest/Tests/WebAssemblyStringComparisonTest.cs:32]: We'd picked up some of the Mono changes and I rebased this PR on top of that.

**Pull request merged**